packages/flare/python/cron_job_ingest_events.py

```diff
@@ -29,46 +31,125 @@ def main():
         raise Exception(str(e))
 
     app: client.Application = splunk_service.apps[APP_NAME]
-    flare_api = FlareAPI(app=app)
+    create_collection(app=app)
+    api_key: Optional[str] = None
+    tenant_id: Optional[int] = None
+    for item in app.service.storage_passwords.list():
+        if item.content.username == "api_key":
+            api_key = item.clear_password
+
+        if item.content.username == "tenant_id":
+            tenant_id = (
+                int(item.clear_password) if item.clear_password is not None else None
+            )
 
-    from_ = get_from_value(app=app)
-    event_feed = flare_api.retrieve_feed(from_=from_)
-    set_from_value(app=app, next=event_feed["next"])
+    if not api_key:
+        raise Exception("API key not found")
 
-    if event_feed["items"]:
-        for item in event_feed["items"]:
-            print(json.dumps(item))
+    if not tenant_id:
+        raise Exception("Tenant ID not found")
+    flare_api = FlareAPI(app=app, api_key=api_key, tenant_id=tenant_id)
 
+    next = get_next(app=app, tenant_id=tenant_id)
+    start_date = get_start_date(app=app)
+    while True:
+        event_feed = flare_api.retrieve_feed(next=next, start_date=start_date)
+        save_start_date(app=app, tenant_id=tenant_id)
+        save_next(app=app, tenant_id=tenant_id, next=event_feed["next"])
+
+        if event_feed["items"]:
+            for item in event_feed["items"]:
+                print(json.dumps(item))
+
+        if (
+            int(datetime.now().timestamp() - cron_job_start_timestamp)
+            >= CRON_JOB_MAX_RUNNING_SECONDS
+        ):
+            break
+        if event_feed["next"]:
+            next = event_feed["next"]
+        else:
+            break
 
-def get_from_value(app: client.Application) -> Optional[str]:
-    from_: Optional[str] = None
+
+def get_next(app: client.Application, tenant_id: int) -> Optional[str]:
```

Comment:
It looks like you are copying the kvstore.data.query() pattern 3 times in this file. Can you extract it in a function?